File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    var answer = [];
    
    for (let i = 0; i < food.length; i++) {
        food[i] = Math.floor(food[i] / 2)
        while (food[i]) {
            answer.push(i)
            food[i] -= 1
        }
    }
    answer = answer.join("") + '0' + answer.reverse().join("")
    
    
    return answer;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    
    for i in range(len(food)):
        if food[i] > 1:
            food[i] = food[i] // 2 
        elif food[i] <= 1:
            food[i] = 0
     
    arr = []
    for i in range(len(food)):
        while food[i]:
            arr.append(str(i))
            food[i] -= 1
    
    answer = "".join(arr)  + "0"
    arr.reverse()
    answer += "".join(arr)
    
    return answer